openmc/data/multipole.py

```diff
@@ -0,0 +1,535 @@
+            for i_poly in range(self.fit_order+1):
+                sig_s += (self.curvefit[i_window, i_poly, _FIT_S]
+                          * broadened_polynomials[i_poly])
+                sig_a += (self.curvefit[i_window, i_poly, _FIT_A]
+                          * broadened_polynomials[i_poly])
+                if self.fissionable:
+                    sig_f += (self.curvefit[i_window, i_poly, _FIT_F]
+                              * broadened_polynomials[i_poly])
+        else:
+            temp = invE
+            for i_poly in range(self.fit_order+1):
+                sig_s += self.curvefit[i_window, i_poly, _FIT_S] * temp
+                sig_a += self.curvefit[i_window, i_poly, _FIT_A] * temp
+                if self.fissionable:
+                    sig_f += self.curvefit[i_window, i_poly, _FIT_F] * temp
+                temp *= sqrtE
+
+        # ======================================================================
+        # Add the contribution from the poles in this window.
+
+        if sqrtkT == 0.0:
+            # If at 0K, use asymptotic form.
+            for i_pole in range(startw, endw):
+                psi_chi = -1j / (self.data[i_pole, _MP_EA] - sqrtE)
+                c_temp = psi_chi / E
+                sig_s += (self.data[i_pole, _MP_RS] * c_temp).real
+                sig_a += (self.data[i_pole, _MP_RA] * c_temp).real
+                if self.fissionable:
+                    sig_f += (self.data[i_pole, _MP_RF] * c_temp).real
+
+        else:
+            # At temperature, use Faddeeva function-based form.
+            dopp = self.sqrtAWR / sqrtkT
+            for i_pole in range(startw, endw):
+                Z = (sqrtE - self.data[i_pole, _MP_EA]) * dopp
+                w_val = _faddeeva(Z) * dopp * invE * sqrt(pi)
+                sig_s += (self.data[i_pole, _MP_RS] * w_val).real
+                sig_a += (self.data[i_pole, _MP_RA] * w_val).real
+                if self.fissionable:
+                    sig_f += (self.data[i_pole, _MP_RF] * w_val).real
+
+        return sig_s, sig_a, sig_f
+
+    def __call__(self, E, T):
+        """Compute scattering, absorption, and fission cross sections.
+
+        Parameters
+        ----------
+        E : Real or Iterable of Real
+            Energy of the incident neutron in eV.
+        T : Real
+            Temperature of the target in K.
+
+        Returns
+        -------
+        3-tuple of Real or 3-tuple of numpy.ndarray
+            Total, absorption, and fission microscopic cross sections at the
+            given energy and temperature.
+
+        """
+
+        fun = np.vectorize(lambda x: self._evaluate(x, T))
+        return fun(E)
+
+    def export_to_hdf5(self, path, mode='a', libver='earliest'):
+        """Export windowed multipole data to an HDF5 file.
+
+        Parameters
+        ----------
+        path : str
+            Path to write HDF5 file to
+        mode : {'r', r+', 'w', 'x', 'a'}
```

Comment:
This function writes to a file and therefore must provide write access. Would this still be able to write if the caller of this method passed `r` as the mode?